scripts/core/editor3/components/Editor3.jsx

```diff
@@ -109,46 +166,54 @@ export class Editor3 extends React.Component {
                 handleKeyCommand={this.handleKeyCommand}
                 onChange={this.onChange}
                 onTab={this.onTab}
-                readOnly={this.readOnly}
+                readOnly={this.props.readOnly}
                 ref="editor"
             />;
 
-        if (this.showToolbar) {
+        const spellchecker = () =>
+            <Spellchecker
```

Comment:
Then this becomes:
```js
<Spellchecker
    spellchecker={this.props.spellchecker}
    editorState={this.state.editorState}
    editorRect={editorRect}
    onChange={this.onChange} />
```
Similar to the toolbar. I'm not sure if sharing the `editorState` is the best pattern, but this is what I've seen as being used in the `draft-js-plugins` project and seems like a decent approach.